includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -1973,7 +1973,9 @@ private function remove_node( $node ) {
 		if ( $node && $parent ) {
 			$this->remove_invalid_child( $node );
 		}
-		while ( $parent && ! $parent->hasChildNodes() && $this->root_element !== $parent ) {
+
+		// @todo Does this parent removal even make sense anymore?
```

Comment:
Are you referring to alternatively not removing empty parent nodes?

Reply:
Yes, that's right. We should evaluate why this was added in the first place, and consider discontinuing the practice. 

I believe it may have been due to having input like this:

```
<p><script>evil</script></p>
```

Normally this would sanitized as:

```
<p></p>
```

That results in what appears to be a blank area in the page. So apparently it helps in that case. But we should consider being more limited in what we consider to be suitable for removal. For example, perhaps this only makes sense for `p` elements.